core/src/main/kotlin/org/evomaster/core/problem/external/service/param/ResponseParam.kt

```diff
@@ -2,23 +2,19 @@ package org.evomaster.core.problem.external.service.param
 
 import org.evomaster.core.problem.api.service.param.Param
 import org.evomaster.core.search.gene.EnumGene
-import org.evomaster.core.search.gene.ObjectGene
+import org.evomaster.core.search.gene.Gene
 import org.evomaster.core.search.gene.OptionalGene
 
-
-class ResponseParam (
-    /**
-     * Contains the values for HTTP status codes
-     */
-    val status: EnumGene<Int> = EnumGene("status", listOf(200, 400, 401, 403, 404, 500)),
-    /**
-     * Response content type, for now supports only JSON
-     */
-    val responseType: EnumGene<String> = EnumGene("responseType", listOf("JSON")),
-    val response: OptionalGene = OptionalGene("response", ObjectGene("response", listOf()))
-        ): Param("response", mutableListOf(status).plus(responseType).toMutableList()) {
-
-    override fun copyContent(): Param {
-        return ResponseParam(status.copy() as EnumGene<Int>, responseType.copy() as EnumGene<String>, response.copy() as OptionalGene)
-    }
-}
\ No newline at end of file
+/**
+ * represent external service responses which contains
+ * @property responseType the data type of the response
+ * @property response represents the response to return
+ * @property extraProperties represent the extra properties to describe the responses, such as status code for HTTP response,
+ * note that all [extraProperties] are considered as part of children of this param.
+ */
+abstract class ResponseParam(
+    name: String,
+    val responseType: EnumGene<String>,
+    val response: OptionalGene,
+    val extraProperties: List<Gene>
```

Comment:
hmmm unsure about this... shouldn't they be explicitely marked here, eg a gene for status code. otherwise, how are u going to infer their phenotype when computing fitness and generating JUnit tests?

Reply:
this is the general response for any API. the HTTP response is identified with `HttpWsResponseParam.kt`

Reply:
i m still unsure. let's discuss it next time we meet